lodash.js

```diff
@@ -2494,12 +2494,17 @@
      *
      * @private
      * @param {Array} array The array to flatten.
-     * @param {boolean} [isDeep] Specify a deep flatten.
+     * @param {number} [depth=1] The maximum recursion depth.
      * @param {boolean} [isStrict] Restrict flattening to arrays-like objects.
      * @param {Array} [result=[]] The initial result value.
      * @returns {Array} Returns the new flattened array.
      */
-    function baseFlatten(array, isDeep, isStrict, result) {
+    function baseFlatten(array, depth, isStrict, result) {
+      depth = isNaN(toNumber(depth)) ? 1 : toInteger(depth);
```

Comment:
Move `toInteger` use to `flattenDeep` to keep it out of `baseFlatten`. The `baseFlatten` should assume receiving an integer depth of > 0. In `flattenDeep` no need for the `isNaN` and `toNumber` juggle just use `toInteger`.